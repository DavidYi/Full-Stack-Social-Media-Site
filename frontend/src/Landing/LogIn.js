import React from 'react';

import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Alert from 'react-bootstrap/Alert';
import Col from 'react-bootstrap/Col';

import {Formik} from 'formik';
import * as yup from 'yup';

import Cookies from 'universal-cookie';

import {Redirect} from "react-router-dom";


class LogIn extends React.Component{
	constructor(props) {
		super(props);
		this.state = {
			isAuth: false,
			showAlert: false,
			username: "",
		}
		this.cookies = new Cookies();
		this.schema = yup.object().shape({
			username: yup.string().required(),
			password: yup.string().required(),
		});
		this.defaultValues  = {
			username: "",
			password: "",
		};


	}

	componentDidMount() {
		//if already logged in, check and then redirect
		if (!this.state.isAuth){
			fetch("https://yo-backend-final.herokuapp.com/login", {
				headers: {
					Accept: 'application/json',
					'Content-Type' : 'application/json'
				},
				body: JSON.stringify({isLoggedIn: true}),
				credentials: 'include',
				method: 'POST'
			}).then( response => response.json())
			.then(json => {
				if (json.redirect){
					this.setState({isAuth: true, username: json.username});
					return;
				}
			})
		}
	}

	render() {
		const handleSubmit = async (values, actions) => {

			const credentials = {
				username: values.username,
				password: values.password
			}
			
			await fetch("https://yo-backend-final.herokuapp.com/login", {
				headers: {
					Accept: 'application/json',
					'Content-Type' : 'application/json'
				},
				credentials: 'include',
				method: 'POST',
				body: JSON.stringify(credentials)
			}).then( response => response.json())
			.then(json => {
				if (json.result === "success")
					this.setState({isAuth: true, username: json.username})
				else
					this.setState({showAlert: true});
			})
			.catch((err) => {
				this.setState({showAlert: true});
			});
		};

		return (
				<div>
				{this.state.showAlert && 
					<Alert variant="danger" onClose={() => this.setState({showAlert: false})} dismissible>
				        <p>
				          Wrong Credentials
				        </p>
			        </Alert>
				}

				<Formik 
					validationSchema={this.schema}
					onSubmit={handleSubmit}
					initialValues={this.defaultValues}
				>
				{({
					handleSubmit,
					handleChange,
					values,
					isSubmitting,
					errors,
				}) => (
					<Form onSubmit={handleSubmit}>
						<Form.Row>
							<Form.Group as={Col} controlId="logUsername">
								<Form.Control 
								type="username" 
								name="username"
								onChange={handleChange}
								isInvalid={!!errors.username}
								value={values.username}
								placeholder="Enter username" />
							</Form.Group>
							<Form.Group as={Col} controlId="logPass">
								<Form.Control 
								type="password"	
					      		name="password"
								onChange={handleChange}
								isInvalid={!!errors.password} 
								value={values.password}
								placeholder="Password" />
							</Form.Group>
							<Form.Group as={Col} controlId="logIn">
								<Button variant="primary" type="submit" disabled={isSubmitting}>
				      			Log In
			      				</Button>
							</Form.Group>
						</Form.Row>
					</Form>
				)}
				</Formik>
					{this.state.isAuth && 
						<div>
							<Redirect to={{
				              pathname: "/main",
				              state: {
				              	loggedIn: this.state.username,
				              }
				            }}/>
			            </div>
		    		}
				</div>
			);
	}
}

export default LogIn;