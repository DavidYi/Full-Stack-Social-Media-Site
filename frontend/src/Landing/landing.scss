.less{
	margin: auto;
	padding: auto;
	width: 50vw;
	
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
}

.bg-landing{
	background: #34495e;
	position: relative;
	min-height: 100vh;
	height: calc(100%);
}