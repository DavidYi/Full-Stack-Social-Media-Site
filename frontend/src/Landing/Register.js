import React, {useState} from 'react';

import Alert from 'react-bootstrap/Alert';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import { Formik, useField, useFormikContext} from 'formik';
import * as yup from 'yup';
import './landing.scss';

const defaultValues = {
	email: "",
	name: "",
	zip: "",
	phone: "",
	password: "",
	passCmf: "",
	dob: "",
}

const schema = yup.object().shape({
	email: yup.string().email('Invalid email').required('Email is required'),
	name: yup.string().required('Display Name is required'),
	zip: yup.string().matches(/^[0-9]{5}(-[0-9]{4})?$/, "Invalid Zip Code").required('Zip Code is required'),
	//phone: yup.string().matches(/^(\+?1)?\(?[0-9]{3}\)?-?[0-9]{3}-?[0-9]{4}$/, 'Invalid phone nunber').required(),
	password: yup.string().required('Password is required'),
	passCmf: yup.string().test('passwords-match', "Passwords don't match!", function(value){
			return this.parent.password === value;
		}).required(''),
	dob: yup.string().required('Date of Birth is required.')
});

const DatePickerField = ({ ...props }) => {
	const { setFieldValue } = useFormikContext();
	const [field] = useField(props);

    return (
		<DatePicker
			{...field}
			{...props}
			dateFormat="MM/dd/yyyy"
			className="form-control"
			selected={(field.value && new Date(field.value)) || null}
			onChange={val => {
				setFieldValue(field.name, val);
			}}
		/>
  );
};

function Register() {
	const [isAuth, setAuth] = useState(false);

	const handleSubmit = async (values, actions) => {
		let formValues = await Object.assign({}, values);
		
		var newUser = {
			username: formValues.name,
			email: formValues.email,
			zipcode: formValues.zip,
			dob: formValues.dob.getTime(),
			password: formValues.password
		}

		await fetch("https://yo-backend-final.herokuapp.com/register", {
			method: 'POST',
			headers: {
				'Content-Type' : 'application/json'
			},
			credentials: 'include',
			body: JSON.stringify(newUser)
		}).then(res => res.json())
		.then(json => {
			if (json.result === "success"){
				actions.setSubmitting(false);
				setAuth(true);
			}
		});
		actions.setSubmitting(false);
	};


	return (
		<Jumbotron className="less">

		{isAuth &&
			<Alert variant="success" onClose={() => setAuth(false)} dismissible>
		        <p>
		          Registration Complete
		        </p>
	        </Alert>}
		<h1>Register</h1>
		<Formik 
			validationSchema={schema}
			onSubmit={handleSubmit}
			initialValues={defaultValues}
		>
		{({
			handleSubmit,
			handleChange,
			handleBlur,
			setFieldValue,
			values,
			touched,
			isSubmitting,
			isValid,
			errors,
		}) => (
			
			<Form noValidate onSubmit={handleSubmit}>
				<Form.Group controlId="registerEmail">
					<Form.Label>Email address</Form.Label>
					<Form.Control
						type="email" 
						name="email"
						onChange={handleChange}
						isInvalid={!!errors.email}
						value={values.email}
						placeholder="Enter email"
					 />
					 <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
					 <Form.Control.Feedback type="invalid">
					 	{errors.email}
					 </Form.Control.Feedback>
		      	</Form.Group>

	      		<Form.Group controlId="registerName">
		      		<Form.Label>Username</Form.Label>
		      		<Form.Control  
			      		type="text" 
						name="name"
						onChange={handleChange}
						isInvalid={!!errors.name}
						value={values.name}
			      		placeholder="Enter display Name" />
		      		<Form.Control.Feedback>Looks good!</Form.Control.Feedback>
		      		<Form.Control.Feedback type="invalid">
					 	{errors.name}
					 </Form.Control.Feedback>
	      		</Form.Group>

	      		<Form.Group controlId="registerDob">
		      		<Form.Label>Date of Birth</Form.Label>
	      			<DatePickerField
                        name="dob"
                        value={values.dob}
						isInvalid={!!errors.dob}
                        onChange={setFieldValue}
                    />
                    <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
		      		<Form.Control.Feedback type="invalid">
					 	{errors.dob}
					 </Form.Control.Feedback>
	      		</Form.Group>

	      		{/*<Form.Group controlId="registerPhone">
		      		<Form.Label>Phone Number</Form.Label>
		      		<Form.Control 
		      		type="text"  
					name="phone"
					onChange={handleChange}
					isInvalid={!!errors.phone}
					value={values.phone}
		      		placeholder="111-111-1111" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.phone}
		      		</Form.Control.Feedback>
	      		</Form.Group>*/}

	      		<Form.Group controlId="registerZip">
		      		<Form.Label>Zip Code</Form.Label>
		      		<Form.Control
		      		type="text" 
		      		name="zip"
					onChange={handleChange}
					isInvalid={!!errors.zip}
					value={values.zip}
		      		placeholder="Zip" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.zip}
		      		</Form.Control.Feedback>
	      		</Form.Group>

	      		<Form.Group controlId="registerPass">
		      		<Form.Label>Password</Form.Label>
		      		<Form.Control
		      		type="password"
		      		name="password"
					onChange={handleChange}
					isInvalid={!!errors.password}
					value={values.password}
		      		placeholder="Password" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.password}
		      		</Form.Control.Feedback>
	      		</Form.Group>

	      		<Form.Group controlId="registerPassCmf">
		      		<Form.Label>Confirm Password</Form.Label>
		      		<Form.Control
		      		type="password"
		      		name="passCmf"
					onChange={handleChange}
					isInvalid={!!errors.passCmf}
					value={values.passCmf}	
		      		placeholder="Confirm Password" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.passCmf}
		      		</Form.Control.Feedback>
	      		</Form.Group>

	      		<Button variant="primary" type="submit" disabled={isSubmitting}>
	      			Register
      			</Button>

	      	</Form>
	      	)}
		</Formik>
		</Jumbotron>
		);
}

export default Register;