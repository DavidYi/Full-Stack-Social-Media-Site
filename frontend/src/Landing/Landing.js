import React from 'react';
import Register from './Register';
import './landing.scss';
import LogIn from './LogIn';
import Header from '../Shared/Header';
import 'bootstrap/dist/css/bootstrap.min.css';

function Landing() {
  return (
  	<div>

  		<Header component={<LogIn />}/>
    <div className="bg-landing">
    	<Register />
    </div>
    </div>
  );
}

export default Landing;
