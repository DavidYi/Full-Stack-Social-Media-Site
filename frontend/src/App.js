import React from 'react';
import './App.scss';
import { Route, Switch, BrowserRouter } from 'react-router-dom'
import Landing from './Landing/Landing';
import MainWrapper from './Main/MainWrapper';
import ProfileWrapper from './Profile/ProfileWrapper';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {
  return (
  	<div>

	<BrowserRouter>
		<Switch>
			<Route exact path="/" component={Landing}/>
			<Route path="/main" component={MainWrapper}/> 
			<Route path="/profile" component={ProfileWrapper}/> 
		</Switch>

	</BrowserRouter>
	</div>
  );
}

export default App;
