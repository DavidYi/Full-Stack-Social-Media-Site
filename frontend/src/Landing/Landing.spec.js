import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Landing from './Landing';
import App from '../App';
import LogIn from './LogIn';
import Register from './Register';

import {Redirect} from "react-router-dom";
import Alert from 'react-bootstrap/Alert';

describe('Landing', () => {
	const container = mount(<App />)
	test('snapshot renders', () => {
		const component = renderer.create(<Landing />);
	    let tree = component.toJSON();
	    expect(tree).toMatchSnapshot();
	});

	it('should have 2 email fields', () => {
    	expect(container.find('input[type="email"]').length).toEqual(2);
  	});

  	it('should have 3 password fields', () => {
    	expect(container.find('input[type="password"]').length).toEqual(3);
  	});

  	it('should have 2 forms', () => {
    	expect(container.find('form').length).toEqual(2);
  	});

});


describe('Register', () => {
	const container = mount(shallow(<Register />).get(0));

	it('renders the inner LogIn Component', () => {
		const wrapper = mount(<App />);
		expect(wrapper.find(Register).length).toEqual(1);
	});

	it('should have email field', () => {
    	expect(container.find('input[type="email"]').length).toEqual(1);
  	});

  	it('should have display name field', () => {
    	expect(container.find('input[name="name"]').length).toEqual(1);
  	});

  	it('should have zipcode field', () => {
    	expect(container.find('input[name="zip"]').length).toEqual(1);
  	});

  	it('should have phone field', () => {
    	expect(container.find('input[name="phone"]').length).toEqual(1);
  	});


  	it('should have 2 password fields (password and passcmf)', () => {
    	expect(container.find('input[type="password"]').length).toEqual(2);
    	expect(container.find('input[name="password"]').length).toEqual(1);
    	expect(container.find('input[name="passCmf"]').length).toEqual(1);
  	});

  	it('should have 1 submit button', () => {
    	expect(container.find('button').length).toEqual(1);
  	});



	it("renders alert that registration is successful when registration is all right otherwise nothing", () => {
		const wrapper  = mount(shallow(<LogIn />).get(0));

		const email = wrapper.find('input').at(0);
		const name = wrapper.find('input').at(1);
		const phone = wrapper.find('input').at(2);
		const zip = wrapper.find('input').at(3);
		const password = wrapper.find('input[name="password"]');
		const passCmf = wrapper.find('input[name="passCmf"]');
		const submit = wrapper.find('button');
		email.instance().value = 'davidcyyi@rice.edu';
		password.instance().value = '1';
		passCmf.value = '12';

		submit.simulate('click');

		setTimeout(() => {
		expect(wrapper.find(Alert).length).toEqual(0);}, 1000);


		name.instance().value = 'hey';
		phone.value = "222-111-1111";
		zip.value = "12345";

		setTimeout(() => {
		expect(wrapper.find(Alert).length).toEqual(0);}, 1000);

		passCmf.value = '1';


		setTimeout(() => {expect(wrapper.find(Redirect).length).toEqual(0);
		expect(wrapper.find(Alert).length).toEqual(1);}, 1000);



	});

	it("should alert if wrong credentials", () => {
		const wrapper  = mount(shallow(<LogIn />).get(0));

		const email = wrapper.find('input').at(0);
		const password = wrapper.find('input').at(1);
		const submit = wrapper.find('button');
		email.instance().value = 'Sincere@april.biz';
		password.instance().value = 'Kulas Light1';

		submit.simulate('click');


		setTimeout(() => {expect(wrapper.find(Alert)).toBeTruthy();
		expect(wrapper.find(Alert).length).toEqual(1);}, 1000);
	});

});

describe('LogIn', () => {
	const container = mount(<LogIn />);

	it('renders the inner LogIn Component', () => {
		const wrapper = mount(<App />);
		expect(wrapper.find(LogIn).length).toEqual(1);
	});

	it('should have email field', () => {
    	expect(container.find('input[type="email"]').length).toEqual(1);
  	});

  	it('should have password field', () => {
    	expect(container.find('input[type="password"]').length).toEqual(1);
  	});

  	it('should have 1 submit button', () => {
    	expect(container.find('button').length).toEqual(1);
  	});


  	it('should have be set to false', () => {
    	expect(container.state('isAuth')).toEqual(false);
    	expect(container.state('showAlert')).toEqual(false);
    	expect(container.state('userInfo')).toEqual({});
  	});


	it("renders Redirect when user autheticated", () => {
		const wrapper  = mount(<LogIn />);

		const email = wrapper.find('input').at(0);
		const password = wrapper.find('input').at(1);
		const submit = wrapper.find('button');
		email.instance().value = 'Sincere@april.biz';
		password.instance().value = 'Kulas Light';

		submit.simulate('click');


		setTimeout(() => {
			expect(wrapper.find(Redirect)).toBeTruthy();
			expect(wrapper.find(Redirect).length).toEqual(1);
			expect(wrapper.state.isAuth).toEqual(true);
		}, 1000);
	});

	it("should alert if wrong credentials", () => {
		const wrapper  = mount(<LogIn/>);

		const email = wrapper.find('input').at(0);
		const password = wrapper.find('input').at(1);
		const submit = wrapper.find('button');
		email.instance().value = 'Sincere@april.biz';
		password.instance().value = 'Kulas Light1';

		submit.simulate('click');

		expect(wrapper.state('isAuth')).toEqual(false);
		setTimeout(() => {
			expect(wrapper.find(Alert)).toBeTruthy();
			expect(wrapper.find(Alert).length).toEqual(1);
			expect(wrapper.state.showAlert).toEqual(true);
		}, 1000);
	});

});