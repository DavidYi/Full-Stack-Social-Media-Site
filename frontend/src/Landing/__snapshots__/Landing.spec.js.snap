// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing snapshot renders 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      Yo
    </span>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <div>
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-row"
          >
            <div
              className="form-group col"
            >
              <input
                className="form-control"
                id="logEmail"
                name="email"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group col"
            >
              <input
                className="form-control"
                id="logPass"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group col"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                Log In
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </nav>
  <div
    className="bg-landing"
  >
    <div
      className="less jumbotron"
    >
      <h1>
        Register
      </h1>
      <form
        className=""
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerEmail"
          >
            Email address
          </label>
          <input
            className="form-control"
            id="registerEmail"
            name="email"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
          <div
            className="valid-feedback"
          >
            Looks good!
          </div>
          <div
            className="invalid-feedback"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerName"
          >
            Display Name
          </label>
          <input
            className="form-control"
            id="registerName"
            name="name"
            onChange={[Function]}
            placeholder="Enter display Name"
            type="text"
            value=""
          />
          <div
            className="valid-feedback"
          >
            Looks good!
          </div>
          <div
            className="invalid-feedback"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerPhone"
          >
            Phone Number
          </label>
          <input
            className="form-control"
            id="registerPhone"
            name="phone"
            onChange={[Function]}
            placeholder="111-111-1111"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerZip"
          >
            Zip Code
          </label>
          <input
            className="form-control"
            id="registerZip"
            name="zip"
            onChange={[Function]}
            placeholder="Zip"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerPass"
          >
            Password
          </label>
          <input
            className="form-control"
            id="registerPass"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <div
            className="invalid-feedback"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="registerPassCmf"
          >
            Confirm Password
          </label>
          <input
            className="form-control"
            id="registerPassCmf"
            name="passCmf"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <div
            className="invalid-feedback"
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;
